### ISSUE TITLE
Cannot build rust_test: Error listing dir `none//buck2/platform/build_mode`

### ISSUE THREAD

**Issue by MorganR** [facebook/buck2#420]
Hello, I have managed to get Buck2 working to build and run a simple local rust library and associated binary.

However, I'm unable to compile the `rust_test` rule to run my tests. 

I get this error:

```
Error running analysis for `root//lib:rs-wordle-solver_test (prelude//platforms:default#524f8da68ea2a374)`

Caused by:
    0: Error looking up configured node root//lib:rs-wordle-solver_test (prelude//platforms:default#524f8da68ea2a374)
    1: Error looking up configured node none//buck2/platform/build_mode:build_mode (prelude//platforms:default#524f8da68ea2a374)
    2: looking up unconfigured target node `none//buck2/platform/build_mode:build_mode`
    3: Error loading targets in package `none//buck2/platform/build_mode` for target `none//buck2/platform/build_mode:build_mode`
    4: Error gathering package listing for `none//buck2/platform/build_mode`
    5: The package is invalid
    6: This error was caused by the end user
    7: Error listing dir `none//buck2/platform/build_mode`
    8: Error listing directory
    9: read_dir(.../rs-wordle-solver/none/buck2/platform/build_mode)
   10: No such file or directory (os error 2)
```

The associated BUCK file is [here](https://github.com/MorganR/rs-wordle-solver/blob/buck/lib/BUCK). I haven't been able to find any information on setting a `build_mode` for rust. Any ideas what could be causing this?

**Comment by ndmitchell** [comment]
Oops. We broke this when adding an internal feature/configuration to build tests on remote execution. We'll fix it and add a test so it doesn't break in future (we've got open source tests for `rust_library` and `rust_binary`, but guess what we're missing thus far... CC @themarwhal). In the meantime, you can fix it with the diff:

```diff
diff --git a/buck2/prelude/decls/common.bzl b/buck2/prelude/decls/common.bzl
--- a/buck2/prelude/decls/common.bzl
+++ b/buck2/prelude/decls/common.bzl
@@ -235,7 +235,7 @@
     )
 
 def _re_action_key_provider_arg() -> Attr:
-    return attrs.dep(providers = [BuildModeInfo], default = "fbcode//buck2/platform/build_mode:build_mode")
+    return attrs.option(attrs.dep(providers = [BuildModeInfo]), default = None)
 
 def _re_test_args() -> dict[str, Attr]:
     return {
```

CC @snufkinsnorka. Why do we have `remote_execution_action_key_providers` both on the rules _and_ on the execution platform? Isn't the execution platform enough? If not, we should probably use the if open source conditional to make the rule switch to the variant above in open source.

**Comment by krallin** [comment]
Neil: test execution doesn’t use execution platforms. Ideally, they would
(well, rather they’d use something similar because the “platform” aspect
isn’t one that really makes sense there), but given they don’t that’s why
it ends up showing in both places

Having one level of indirection between ExternalRunnerTestInfo and it’s
execution platforms would be desirable

(Agreed it should be made optional in the meantime)

**Comment by MorganR** [comment]
Thank you for the fast response, folks! That has worked nicely. Should I close this, or do you want to keep it open until the issue is fixed at head?

**Comment by ndmitchell** [comment]
Definintely needs fixing in head! And a test. Going to do that now.

**Comment by ndmitchell** [comment]
All done and works with a demo - just need someone internal to approve my diffs and it will go out.

**Comment by ndmitchell** [comment]
Fixed in cd05fd064ec2db80de188ed5a0b766826396bf9f. Hopefully this will never break again!
